### PR TITLE
Refactoring/305 rewrite hive integration test

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**

### CODE REVIEW

File: src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java [hunk 1]
```diff
@@ -1,385 +1,652 @@
...
-                + ".t2 b ON a.x=b.x";
-        final ResultSet result = executeQuery(query);
-        assertAll(() -> assertNextRow(result, 2L, "bbb", 2L, "bbb"), () -> assertFalse(result.next()));
+    private static String getPropertyFromFile(final String filePath, final String propertyName) {
+        try (final InputStream inputStream = new FileInputStream(filePath)) {
+            final Properties properties = new Properties();
+            properties.load(inputStream);
+            return properties.getProperty(propertyName);
+        } catch (final IOException exception) {
+            throw new IllegalArgumentException("Cannot access the oracle.properties file or read from it.");
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java]: Turning an `IOException` into an `IllegalArgumentException` makes no sense. Simply throw the IOException. It's test code after all.

File: src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java [hunk 2]
```diff
@@ -1,385 +1,652 @@
...
+            throws InterruptedException, BucketAccessException, TimeoutException {
+        final Bucket bucket = exasolContainer.getDefaultBucket();
+        bucket.uploadFile(PATH_TO_VIRTUAL_SCHEMAS_JAR, VIRTUAL_SCHEMAS_JAR_NAME_AND_VERSION);
+        bucket.uploadFile(Path.of(driverPath, driverName), "drivers/jdbc/" + driverName);
+        bucket.uploadFile(HIVE_DRIVER_SETTINGS_PATH, "drivers/jdbc/" + JDBC_DRIVER_CONFIGURATION_FILE_NAME);
+    }
+
+    private static void createTableHiveSimple(final Statement statementHive) throws SQLException {
+        statementHive.execute("CREATE TABLE " + TABLE_HIVE_SIMPLE + "(X INT)");
+        statementHive.execute("TRUNCATE TABLE " + TABLE_HIVE_SIMPLE);
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java]: Why do you truncate immediately after creating the table?

File: src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java [hunk 3]
```diff
@@ -1,385 +1,652 @@
...
+        statementHive.execute("INSERT INTO " + TABLE_HIVE_SIMPLE + " VALUES (99)");
+    }
+
+    private static void createTableDecimalCast(final Statement statementHive) throws SQLException {
+        statementHive.execute("CREATE TABLE " + TABLE_HIVE_DECIMAL_CAST
+                + "(DECIMAL_COL1 DECIMAL(12, 6), DECIMAL_COL2 DECIMAL(36, 16), DECIMAL_COL3 DECIMAL(38, 17))");
+        statementHive.execute("TRUNCATE TABLE " + TABLE_HIVE_DECIMAL_CAST);
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java]: dto.

File: src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java [hunk 4]
```diff
@@ -1,385 +1,652 @@
...
 
     @Test
-    void testLeftJoin() throws SQLException {
-        final String query = "SELECT * FROM  " + VIRTUAL_SCHEMA + ".t1 a LEFT OUTER JOIN  " + VIRTUAL_SCHEMA
-                + ".t2 b ON a.x=b.x ORDER BY a.x";
-        final ResultSet result = executeQuery(query);
-        assertAll(() -> assertNextRow(result, 1L, "aaa", null, null), () -> assertNextRow(result, 2L, "bbb", 2L, "bbb"),
-                () -> assertFalse(result.next()));
-    }
+    public void testSetup() throws SQLException {
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java]: Why is this called `testSetup()`?
What does it really test?

File: doc/development/developing-sql-dialect/integration_testing_with_containers.md [hunk 5]
```diff
@@ -78,13 +78,24 @@ Therefore we cannot include some jdbc drivers to the projects and you need to do
 
 List of disabled integration tests:
 
+* HiveSqlDialectIT
 * OracleSqlDialectIT
 
+How to start `HiveSqlDialectIT`:
+
+1. Download Hive JDBC driver `HiveJDBC41.jar`.
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply doc/development/developing-sql-dialect/integration_testing_with_containers.md]: done

File: doc/development/developing-sql-dialect/integration_testing_with_containers.md [hunk 6]
```diff
@@ -78,13 +78,24 @@ Therefore we cannot include some jdbc drivers to the projects and you need to do
...
+2. Temporarily put it into `src/test/resources/integration/driver/hive` directory.
+3. If the file's name is different from the one mentioned above, edit `src/test/resources/integration/driver/hive/hive.properties` and `settings.cfg` files.
+4. Run the tests from an IDE or temporarily add `HiveSqlDialectIT.java` into the `maven-failsafe-plugin`'s includes section and execute `mvn verify` command.
+5. Remove the driver after the test. Do not upload it to the GitHub repository.
+
 How to start `OracleSqlDialectIT`:
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply doc/development/developing-sql-dialect/integration_testing_with_containers.md]: done

File: src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java [hunk 7]
```diff
@@ -39,17 +39,16 @@
 /*
 * How to run `OracleSqlDialectIT`:
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java]: done

File: src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java [hunk 8]
```diff
@@ -1,385 +1,652 @@
...
+
+import com.exasol.bucketfs.Bucket;
+import com.exasol.bucketfs.BucketAccessException;
+import com.exasol.containers.ExasolContainer;
+import com.exasol.containers.ExasolContainerConstants;
+
+import utils.IntegrationTestSetupManager;
+
+/*
+ * How to run `HiveSqlDialectIT`:
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java]: done

File: src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java [hunk 9]
```diff
@@ -1,385 +1,652 @@
...
+    public static final String HIVE_DOCKER_COMPOSE_YAML = "src/test/resources/integration/driver/hive/docker-compose.yaml";
+    public static final Path HIVE_DRIVER_SETTINGS_PATH = Path.of("src", "test", "resources", "integration", "driver",
+            "hive", JDBC_DRIVER_CONFIGURATION_FILE_NAME);
+    private static final String PATH_TO_HIVE_PROPERTY_FILE = "src/test/resources/integration/driver/hive/hive.properties";
+    private static final String HIVE_SERVICE_NAME = "hive-server_1";
+    private static final int HIVE_EXPOSED_PORT = 10000;
+    private static final String CONNECTION_HIVE_JDBC = "CONNECTION_HIVE_JDBC";
+    private static final String SCHEMA_HIVE = "default";
+    private static final String HIVE_USERNAME = "hive";
+    private static final String HIVE_PASSWORD = "hive";
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java]: It didn't

File: src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java]: done

File: src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java (hunk 2 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java]: I have no idea, it's legacy code

File: src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java (hunk 3 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java]: dto..

File: src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java (hunk 4 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java]: Probably that virtual schemas were created correctly. Also legacy

File: src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java [hunk 10]
```diff
@@ -38,18 +38,14 @@
 
 /*
 * How to run `OracleSqlDialectIT`:
-* Download Oracle JDBC driver `ojdbc8.jar` and oracle instant client `instantclient-basic-linux.x64-12.1.0.2.0.zip`.
-* Temporary put them into `src/test/resources/integration/driver/oracle` directory.
-* If the files' names are different from the ones mentioned above, edit `src/test/resources/integration/driver/oracle/oracle.properties` file.
-* Run the tests from an IDE or temporarily add `OracleSqlDialectIT.java` into the `maven-failsafe-plugin`'s includes section and execute `mvn verify` command.
+* See the documentation: doc/development/developing-sql-dialect/integration_testing_with_containers.md
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java]: Put in a hyper ref please. Same for Hive.

File: src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java (hunk 4 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java]: fixed

File: src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java (hunk 10 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java]: fixed

**Pull request merged**